### PR TITLE
Clara's Airport Challenge Submission

### PR DESCRIPTION
**Pull request by Parradominguezcm**
Pulling the final, clean version of my airport project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,3 +79,45 @@ Planes that have landed must be at an airport
 Your task is to test drive the creation of a set of classes/objects to satisfy all the above user stories. You will need to use a random number generator to set the weather (it is normally sunny but on rare occasions it may be stormy). In your tests, you'll need to stub random behaviour to ensure consistent test behaviour.
 
 Your code should defend against [edge cases](http://programmers.stackexchange.com/questions/125587/what-are-the-difference-between-an-edge-case-a-corner-case-a-base-case-and-a-b) such as inconsistent states of the system ensuring that planes can only take off from airports they are in; planes that are already flying cannot take off and/or be in an airport; planes that are landed cannot land again and must be in an airport, etc.
+
```
Comment (Parradominguezcm) [inline README.md:82]: I need to learn markdown so this looks better, and my domain modeling still needs some work. I wonder if I could start writing them in mindmaps instead of tables.

File: src/airport.js [hunk 2]
```diff
@@ -1 +1,53 @@
...
 
+    constructor(airportCapacity = 10, planesStanding = []) {
+        this.#airportCapacity = airportCapacity;
+        this.#planesStanding = planesStanding;
+    }
+
+    landPlane(plane) {
+        if (!plane?.id) return `Unidentified planes are not allowed at the airport`
+        if (this.fullAirport()) {
+            return `Sorry, ${plane.id}, no room for you to park!`;
```
Comment (Parradominguezcm) [inline src/airport.js:13]: maybe my messages are too airport specific, I need to review how to write them in the index file instead of the class to keep things tidy.

File: src/airport.js [hunk 3]
```diff
@@ -1 +1,53 @@
...
+        } else if (!this.#planesStanding.includes(plane)) {
+            this.#planesStanding.push(plane);
+            return `Welcome to your destination, ${plane.id}`;
+        } else {
+            return `${plane.id} is already here`;
+        }
+    }
+
+    orderTakeOff(plane) {
+        if (this.#planesStanding.includes(plane)) {
```
Comment (Parradominguezcm) [inline src/airport.js:23]: This conditional segment took me forever to develop because I was trying to use the filter method, which I still don't understand very well. I went with a slightly convoluted, but simpler version in the end and I think it looks great! It's easier to read, but the filter would have halved the number of lines.

File: test/airportTestSuite.spec.js [hunk 4]
```diff
@@ -0,0 +1,289 @@
+const { assertEquals } = require("./testing-framework");
+const Airport = require("../src/airport");
+const Plane = require("../src/plane");
+
```
Comment (Parradominguezcm) [inline test/airportTestSuite.spec.js:4]: I think my tests are quite comprehensive! I am looking forward to using jasmine and a linter, as it took forever to format and clear all the tests.

File: test/airportTestSuite.spec.js [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+// Report
+console.log(`Test 9: Planes not at the airport can not leave: ${result}`);
+console.log(`Test 9: ${result ? `PASS` : `FAIL`}`);
+console.log(``);
+
+// Clean Up
+airport.planesStanding = [];
+result = undefined;
+actual = undefined;
+
```
Comment (Parradominguezcm) [inline test/airportTestSuite.spec.js:260]: I am really happy with how much TDD I did, and I hope to continue in my next projects. I find it quite hard sometimes to put the logic down in code, so this is great practice.

**Comment by Parradominguezcm** [comment]
Things I like about the project:
- TDD! I was super strict with myself on the order of test-source writing, I think I learned a lot!
- I kept things simple. It may have meant a couple of extra lines sometimes, but I wrote easily understandable code.
- I believe my tests to be very comprehensive.
Things I'd like to do better:
- I need a deeper understanding of JS functions, in some cases they could have saved me lines of code, I just didn't fully understand how they worked.
- I would have liked to make it so all the returned messages came from the index file. I am not sure it's possible, but it would have left my class looking cleaner.
Goals:
- Learn markdown for better-looking readmes.
- learn to test clearer, and to lint. doing so manually took forever on my small screen.
- write better domain models, I am really struggling with the bank challenge and need a clearer idea of the project.
